mica/archive/obsid_archive.py

```diff
@@ -787,10 +786,10 @@ def update(self):
                 else:
                     # if there are no science_2 entries at all for the obsid
                     # see if it is discarded
-                    with Ska.DBI.DBI(dbi='sybase', server='sqlsao', database='axafocat') as db:
+                    with Sqsh(dbi='sybase', server='sqlsao', database='axafocat') as db:
                         target_status = db.fetchone(
                             "select status from target where obsid = {}".format(obs['obsid']))
-                        if target_status['status'] == 'discarded':
+                        if target_status and target_status['status'] == 'discarded':
```

Comment:
It looks like DBI was also returning None for no rows with `fetchone`, so this code change is really a bugfix not due to any change from Sqsh API .